docs/using_eessi/building_on_eessi.md

```diff
@@ -6,67 +6,113 @@ Building on top of EESSI with EasyBuild is relatively straightforward. One cruci
-
 ### Building
-Now, you are ready to build. For example, at the time of writing, `netCDF-4.9.0-gompi-2022a.eb` was not in the EESSI environment yet, so you can build it yourself:
+Now, you are ready to build. For example, suppose you want to install `netcdf4-python-1.6.5-foss-2023b.eb` (which is not present at the time of writing), you run:
+
 ```
-eb netCDF-4.9.0-gompi-2022a.eb
+eb netcdf4-python-1.6.5-foss-2023b.eb
 ```
 
 !!! Note
-    If this netCDF module is available by the time you are trying, you can force a local rebuild by adding the `--rebuild` argument in order to experiment with building locally, or pick a different EasyConfig to build.
+    If this netCDF for python module is available by the time you are trying, you can force a local rebuild by adding the `--rebuild` argument in order to experiment with building locally, or pick a different EasyConfig to build.
 
 ### Using the newly built module
-First, you'll need to add the subdirectory of the `EASYBUILD_INSTALLPATH` that contains the modules to the `MODULEPATH`. You can do that using:
 
+Our new module is readily available on the `MODULEPATH`:
 ```
-module use ${EASYBUILD_INSTALLPATH}/modules/all
-```
+$ module av netcdf4-python/1.6.5-foss-2023b
 
-you may want to do this as part of your `.bashrc`.
+---- /home/<user>/eessi/versions/2023.06/software/linux/x86_64/amd/zen2/modules/all ----
+   netcdf4-python/1.6.5-foss-2023b
+```
 
-!!! Note
-    Be careful adding to the `MODULEPATH` in your `.bashrc` if you are on a cluster with heterogeneous architectures. You don't want to pick up on a module that was not compiled for the correct architectures accidentally.
+Note however that it is the `EESSI-extend` module that added this to our `MODULEPATH`. If we unload that module, the user-installation is no longer on the `MODULEPATH`
+```
+$ module av netcdf4-python/1.6.5-foss-2023b
+No module(s) or extension(s) found!
+```
 
-Since your module is built on top of the EESSI environment, that needs to be loaded first (as described [here](../using_eessi/setting_up_environment.md)), if you haven't already done so.
+This means you'll _always_ need to load the `EESSI-extend` module if you want to use these modules. For example, if you want to load this module in a job, you should have
 
-Finally, you should be able to load our newly build module:
 ```
-module load netCDF/4.9.0-gompi-2022a
+# Loading EESSI-extend makes the netcdf4-python module available on the MODULEPATH
+module load EESSI-extend/2023.06-easybuild
+module load netcdf4-python/1.6.5-foss-2023b
 ```
 
+in your job script.
+
+!!! Note
+    The `EESSI-extend` module does _not_ add `EESSI_SITE_INSTALL` to the `MODULEPATH`. As a system administrator, you will have to make sure that's on the `MODULEPATH` of your users through other means if you want your users to be able to use module installed by `EESSI-extend`.
```

Comment:
This is not really correct, EESSI initialisation by default adds this location to `MODULEPATH` so if you are using our init script or our modules this will automatically be in your `MODULEPATH`

Reply:
Ah, great. But only if `EESSI_SITE_INSTALL` is set when initializing, I suppose? I'll change the note to that  - make sure that `EESSI_SITE_INSTALL` is set before users initialize EESSI.

Reply:
No, it is always set (but of course usually empty)

Reply:
Yes, this is all correct.

Reply:
and `EESSI_USER_INSTALL` is the default with a location of `$HOME/EESSI`

Reply:
Ok, I rewrote these sections, I think it should be ok now.